index.js

```diff
@@ -0,0 +1,177 @@
+  reserveButton.empty();
+
+  console.log(`${tripDetailUrl}`);
+
+  axios.get(tripDetailUrl)
+    .then((response) => {
+      reportStatus(`Trip details loaded!`);
+      targetTripName = response.data.name;
+      singleTripDetail.append(
+        `<h1>HERE'S ONE GOOD WAY TO GET TREKKED UP:</h1>
+        <section class="details_box" >
+          <h3>Name: ${targetTripName}</h3>
+          <h4>Continent: ${response.data.continent}</h4>
+          <h4>Category:  ${response.data.category}</h4>
+          <h5>Weeks: ${response.data.weeks}</h5>
+          <h5>Cost: ${response.data.cost}</h5>
+          <h5>About: </h5>
+          <p>${response.data.about}</p>
+          <h6>Trip ID: ${targetTripId}</h6>
+        </section>`);
+    })
+    .catch((error) => {
+      reportStatus(`Encountered an error while loading trip details: ${error.message}`);
+      console.log(error);
+    });
+    if (singleTripDetail.length > 0 ) {
+      reserveButton.append(`<button id="reservation">I Would Like To Trek Up in This Manner</button>` )
+    }
+};
+
+
+const callReservationScreen = function callReservationScreen() {
+
+    reportStatus('Pulling up reservation screen');
+
+    const reservationScreen = $('#reservation-box');
+    reservationScreen.empty();
+
+    reservationScreen.append(
+      ` <h1> SECURE YOUR OPPORTUNITY TO TREK UP ROYALLY: </h1>
+        <h3 class="details_box"> The Trip You Want To Go On Is: ${targetTripName}</h3>
+        <form class="details_box" id='reservation-form' name='reservation-form'>
+          <label for='name'>Name:</label>
+          <input type='text' id='name' name='name'><br>
+          <label for='email'>Email:</label>
+          <input type='email' id='email' name='email'><br>
+          <label>&nbsp;</label>
+          <input type='submit' id='submit-reservation-info'
+          value='I hereby commit to unapologetically trekking up my life'><br>
+        </form>`
+    )
+  };
+
+const createReservation = (e) => {
+
+  e.preventDefault()
+
+  console.log('in the Create Reservation method!')
+
+  console.log(`Here is the targetTripId: ${targetTripId}`)
+  console.log(`Here is the targetTripName: ${targetTripName}`)
+
+  const reserveTripUrl = BASE_URL.concat("/").concat(targetTripId).concat("/reservations")
+
+  console.log(`Here is the reservation-posting URL: ${reserveTripUrl}`)
+
+  const reservationResult = $('#result-box');
+  reservationResult.empty();
+
+  const reservationData = {
+    // name: 'Mona VanDerWaal',
+    // email: 'scarysmart@happycrazy.ru'
+    name: $('#name').val(),
+    email: $('#email').val()
+  }
+
+  console.log(`Here is the name for the reservation ${reservationData.name}`)
+  console.log(`Here is the email for the reservation ${reservationData.email}`)
+
+  reportStatus(`Submitting a request to reserve a slot in ${targetTripName}.`);
+
+  axios.post(reserveTripUrl, reservationData)
+    .then((response) => {
+      console.log(response);
+      reservationScreen.empty();
```

Comment:
When I fill out the reservation form correctly to make a reservation, I actually get an unexpected error... that's because `reservationScreen` is undefined at this point, and it kicks it over to the `.catch((error) {})` part.

I fixed it by adding `  const reservationScreen = $('#reservation-box');` before this line